File: backend/models/Document.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/sequelize');

const Document = sequelize.define('Document', {
  id:                   { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  odRequestId:          { type: DataTypes.INTEGER, allowNull: false },
  registrationFormPath: { type: DataTypes.STRING },
  paymentProofPath:     { type: DataTypes.STRING },
  posterPath:           { type: DataTypes.STRING },
  geoTagPhotoPath:      { type: DataTypes.STRING },
  certificatePath:      { type: DataTypes.STRING },
  proofUploadedAt:      { type: DataTypes.DATE }
}, { timestamps: true });

module.exports = Document;

File: springboot-backend/src/main/java/com/example/odsystem/model/Document.java
package com.example.odsystem.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "documents")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Document {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "od_request_id", nullable = false)
    private OdRequest odRequest;

    // Documents submitted at application time
    private String registrationFormPath;
    private String paymentProofPath;
    private String posterPath;

    // Proof documents submitted after OD completion
    private String geoTagPhotoPath;
    private String certificatePath;
    private LocalDateTime proofUploadedAt;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
